### PR TITLE
Trigger go vet, gofmt, and goimports during the actions workflow.

### PR DESCRIPTION
**Pull request by wbxyz**
I also had to fix a go vet finding in our benchmark generation.

### CODE REVIEW

**Comment by wbxyz** [comment]
I added the gofmt changes here too, which also includes a cleanup commit.

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -26,3 +26,9 @@ jobs:
 
     - name: Test
       run: go test -v ./...
+
+    - name: Vet
+      run: go vet -v ./...
+
+    - name: Format
+      run: '[ ! "$(gofmt -s -l .)" ]'
```
Comment (msuozzo) [inline .github/workflows/go.yml:34]: let's also use goimports which i think is just called with `-l .`

File: tools/benchmark/generate/main.go [hunk 2]
```diff
@@ -723,7 +723,7 @@ func main() {
 		}
 		log.Printf("Generating %s...", b.Filename)
 		todo <- nil
-		go func() {
+		go func(b *RebuildBenchmark) {
```
Comment (msuozzo) [inline tools/benchmark/generate/main.go:726]: ugh. this is no longer necessary when we raise the minimum supported go version but it's fine for now.

**Review by msuozzo — APPROVED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (wbxyz) [reply .github/workflows/go.yml:38]: Added.

**Pull request merged**